### DISCUSSION TITLE
Jump directly to the picked component w/o displaying the menu

### DISCUSSION THREAD

**Discussion by aidigital** [Bekaboo/dropbar.nvim#200] [Q&A] [answered]
Hi, great plugin, but after looking through the README I can't see if there's a way to do the following:
- call `require("dropbar.api").pick()` 
- look at the letters that appear, and type the desired one
- immediately jump to that component, without any menu popping up

I saw [here](https://github.com/Bekaboo/dropbar.nvim/issues/66#issuecomment-1664295286) how to make it not expand the component when I click `<CR>` (`Enter`), but that's not enough, I'd like to disable the menu entirely.

**Comment by aidigital** [comment]
Ok for now I solved it by overriding [default_on_click](https://github.com/Bekaboo/dropbar.nvim/blob/321cbc7a6be672f15e3ec5ceaab7a90b4dec6122/lua/dropbar/bar.lua#L84) and adding the below lines at the end.

```lua
local function default_on_click(symbol, _, _, _, _)
  -- ... default code ...
  
  --  click on the component, as per:
  -- https://github.com/Bekaboo/dropbar.nvim/issues/66#issuecomment-1664295286
  local api = require('dropbar.api')
  local menu = api.get_current_dropbar_menu()
  if not menu then
    return
  end
  local cursor = vim.api.nvim_win_get_cursor(menu.win)
  local entry = menu.entries[cursor[1]]
  local component =
      entry:first_clickable(entry.padding.left + entry.components[1]:bytewidth())
  if component then
    menu:click_on(component, nil, 1, 'l')
  end

end
```

This is obviously a hack, it's not ideal because we still do the work of displaying the drop-down which could be avoided.
But if there's no easy way to jump directly when using `pick()` without displaying the drop-down, what is at least a better way to implement the above logic?
The README shows how to provide a custom `on_click` callback, but doesn't seem to allow overriding the default on_click behavior, or maybe I didn't do it properly there.

**Comment by Bekaboo** [comment]
Good idea. I will provide a way to override the default `on_click()` callback in the next several commits.

**Reply by Bekaboo** [reply, marked as answer]
Implemented in commit b63fe2e

**Reply by aidigital** [reply]
Amazing! Thank you @Bekaboo
